libnmstate/nm/connection.py

```diff
@@ -21,26 +21,67 @@
 from . import nmclient
 
 
-def create_profile(settings):
-    con_profile = nmclient.NM.SimpleConnection.new()
-    for setting in settings:
-        con_profile.add_setting(setting)
+class ConnectionProfile(object):
+
+    def __init__(self, profile=None):
+        self._con_profile = profile
+        self._nmclient = nmclient.client()
+        self._mainloop = nmclient.mainloop()
+
+    def create(self, settings):
+        self._con_profile = nmclient.NM.SimpleConnection.new()
+        for setting in settings:
+            self._con_profile.add_setting(setting)
+
+    def import_by_device(self, nmdev):
+        self._con_profile = None
+        ac = get_device_active_connection(nmdev)
+        if ac:
+            self._con_profile = ac.props.connection
+
+    def import_by_id(self, con_id):
+        self._con_profile = None
+        if con_id:
+            self._con_profile = self._nmclient.get_connection_by_id(con_id)
+
+    def update(self, profile):
+        self._con_profile.replace_settings_from_connection(profile)
+
+    def add(self, save_to_disk=True):
```

Comment:
IMHO in the end this should be a method of an NM class (or maybe NMClient), since it is about adding the profile object to NM.

Reply:
I think it is valid to say something like: A profile is adding itself to the system.
There is no one correct solution, both may be correct, but one is easier to use then the other.

Reply:
The current solution requires the class to know about the main loop which it would not if only the client runs the main loop if necessary.

Reply:
The same can be said for the function/method that runs it... It either needs to know about the client or about the object. One of them needs to know about the client.
Both approaches are valid IMO, and if one is better then the other, it can be changed with a proper explanation on why it is better.